### PR TITLE
User notification refactor, add getActiveNotificationsOlderThan, add notified field

### PR DESCRIPTION
**Pull request by joeferner**
- [x] @sfeng88 @rygim @jharwig 
- [x] @srfarley
- [ ] @EvanOxfeld @joeybrk372
- [ ] @mwizeman @dsingley

### CODE REVIEW

File: core/core/src/main/java/org/visallo/core/util/PeriodicBackgroundService.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        t.setDaemon(true);
+        t.setName(getThreadName());
+        t.start();
+    }
+
+    private void runPeriodically() {
+        try {
+            while (enabled) {
+                LOGGER.debug("running periodically");
+                Date now = new Date();
```
Comment (srfarley) [inline core/core/src/main/java/org/visallo/core/util/PeriodicBackgroundService.java]: Using the new `java.time` classes might be better.

File: core/core/src/main/java/org/visallo/core/util/PeriodicBackgroundService.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
+package org.visallo.core.util;
+
+import org.apache.commons.lang.time.DateUtils;
+import org.visallo.core.exception.VisalloException;
+import org.visallo.core.model.lock.LeaderListener;
+import org.visallo.core.model.lock.LockRepository;
+
+import java.util.Date;
+
+public abstract class PeriodicBackgroundService {
```
Comment (srfarley) [inline core/core/src/main/java/org/visallo/core/util/PeriodicBackgroundService.java]: Why not use `java.util.concurrent.ScheduledExecutorService`?

File: core/core-test/src/test/java/org/visallo/core/model/notification/UserNotificationRepositoryTest.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+        assertEquals("Current", activeNotifications.get(0).getTitle());
+    }
+
+    @Test
+    public void testGetActiveNotificationsInDateRange() {
+        List<UserNotification> activeNotifications;
+        long currentTime = new Date().getTime();
+
+        simpleOrmSession.save(
+                new UserNotification("user1", "t-120", "Message 1", null, null, new Date(currentTime - 120000), null),
```
Comment (srfarley) [inline core/core-test/src/test/java/org/visallo/core/model/notification/UserNotificationRepositoryTest.java]: I would prefer using the new `java.time` classes for the `UserNotification` API.

File: core/core/src/main/java/org/visallo/core/util/PeriodicBackgroundService.java (hunk 2 above)
Comment (joeferner) [inline core/core/src/main/java/org/visallo/core/util/PeriodicBackgroundService.java]: This does more than just periodic schedules event. It ensures that only one web server is doing this at a given time. This is also just copied from existing code and any further refactors deserves a new PR

File: core/core/src/main/java/org/visallo/core/util/PeriodicBackgroundService.java (hunk 1 above)
Comment (joeferner) [inline core/core/src/main/java/org/visallo/core/util/PeriodicBackgroundService.java]: again copied from existing code. I think any additional changes should be a new PR

File: core/core-test/src/test/java/org/visallo/core/model/notification/UserNotificationRepositoryTest.java (hunk 3 above)
Comment (joeferner) [inline core/core-test/src/test/java/org/visallo/core/model/notification/UserNotificationRepositoryTest.java]: Are we going to change the API? Again sounds like a new PR

**Comment by srfarley** [comment]
@joeferner I'm OK with a separate PR to upgrade to the new Java time API.

**Pull request merged**